x/session/keeper/keeper.go

```diff
@@ -53,3 +53,33 @@ func NewKeeper(
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
+
+// BeginBlocker is called at the beginning of every block
+// It fetches the block hash from the committed block ans saves its hash
+// in the store.
+func (k Keeper) BeginBlocker(ctx sdk.Context) {
+	// ctx.BlockHeader().LastBlockId.Hash is the hash of the last block committed
+	hash := ctx.BlockHeader().LastBlockId.Hash
+	// ctx.BlockHeight() is the height of the current block being processed,
+	// decrementing it by 1 gives us the height of the last block committed.
+	height := ctx.BlockHeight() - 1
```

Comment:
Why not use `ctx.BlockHeader().Height`? https://github.com/tendermint/tendermint/blob/v0.34.x/spec/core/data_structures.md

I don't think it matters but then we're using a single source.